docs/traffic-mirroring-proposal.md

```diff
@@ -0,0 +1,1282 @@
+                        {
+                            "name": "mirror-2",
+                            "selection": ["src-port"]
+                        }
+                    ]
+                }
+            }
+          ]
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ovs-cons-1
+spec:
+  template:
+    metadata:
+      annotations:
+        k8s.v1.cni.cncf.io/networks: |-
+          [
+            {
+                "name":"ovs-vlan300",
+                "interface":"eth1",
+                "namespace":"emu-cni"
+            },
+            {
+                "name":"ovs-cons-mir",
+                "interface":"eth2",
+                "namespace":"emu-cni",
+                "cni-args": {
+                    "mirrors": [
+                        {
+                            "name": "mirror-1",
+                            "output": "port"
+                        }
+                    ]
+                }
+            },
+            {
+                "name":"ovs-cons-mir",
+                "interface":"eth3",
+                "namespace":"emu-cni",
+                "cni-args": {
+                    "mirrors": [
+                        {
+                            "name": "mirror-2",
+                            "output": "port"
+                        }
+                    ]
+                }
+            }
+          ]
+```
+
+### Pros and Cons - NAD approach VS cni-args approach
+
+NAD approach requires a NAD for every VLAN and, when multiple mirrors are present, if two pods on the same VLAN requires to produce to two different mirrors, you have to define two NADs with the same VLAN ID but different mirror configuration (ref. test-case 2. NAD approach vs cni-args approach). This will result in a high number of NADs (as you can see in the test-cases above).
+
+Instead, in cni-args approach, you can pass mirror configuration via Deployment and use NADs only to define OVS ports and VLANs. Anyway, in most of the cases, pods in the same VLAN ID will produce to one set of mirrors. From a developer perspective, cni-args is much more flexible with respect to NAD approach. 
+
+From an isolation point of view, NAD seems more secure: it limits the resources (the mirrors in this case) a pod can create/use. While with cni-args any pod may potentially request to produce/consume to/from any mirror without constraints. Although, we are not expert in kubernetes resources security/isolation, so the disadvantage we highlighted above may not be a real issue.
+
+It's also possible to create 2 different plugins:
+
+- `ovs-mirror-producer`: it only adds input ports (previously created by `ovs` plugin) into `select_*` columns of `Mirror` table
+- `ovs-mirror-consumer`: it only adds the output into `output_port` and `output_vlan` columns of `Mirror` table
+
+The advantage of having 2 different plugins is an easier input validation, in fact we can define specific JSON keys for a better developer experience. 
```

Comment:
I would be all for introducing `ovs-mirror` or `ovs-mirror-producer` and `ovs-mirror-consumer` as independent executables that are kept under a single project, reusing existing libraries. It would may be a good start to make this project more extendable.